consensus/consensus_test.go

```diff
@@ -256,6 +256,82 @@ func TestAdvanceView(t *testing.T) {
 	fc.Stop()
 }
 
+func TestCaptureValidatorVote(t *testing.T) {
+	oneHourInNanoSecs := time.Duration(3600000000000)
+
+	fc := NewFastHotStuffConsensus()
+
+	// BlockHeight = 1, Current View = 2
+	err := fc.Init(oneHourInNanoSecs, oneHourInNanoSecs, createDummyBlock(), createDummyValidatorSet())
+	require.NoError(t, err)
+
+	// Start the event loop
+	fc.Start()
+
+	// Current View = 3
+	{
+		currentView, err := fc.AdvanceView()
+		require.NoError(t, err)
+		require.Equal(t, uint64(3), currentView)
+	}
+
+	// Test with malformed vote
+	{
+		err := fc.CaptureValidatorVote(nil)
+		require.Error(t, err)
```

Comment:
Do we want to check for specific errors in this test?

Reply:
Done